File: API_SC_01_08Abril/Joao_Paulo/joao_paulo_lopes/q3_media.js
import { GetNumberAbove, GetNumberBetween, input , print} from "./utils.js"

function VerificarMaiorNota(Nota1, Nota2){//função que retorna qual é a maior nota dos alunos

    if(Nota1 > Nota2){
        return Nota1
    }
    return Nota2
}


function VerificarMenorNota(Nota1, Nota2){//função que retorna qual é a menor nota dos alunos
    
    if(Nota1 < Nota2){
        return Nota1
    }
    return Nota2
}


function ComputarPorcentagem(Valor, Total){//função que calcula a porcentagem da performace
    return (Valor * 100) / Total
}


function VerificarDesempenho(Nota){//função que verifica a perfomace do aluno segundo a tabela 

    if(Nota <= 2){
        return 'PÉSSIMO'
    }else if(Nota <= 4){
        return 'RUIM'
    }else if(Nota < 7){
        return 'REGULAR'
    }else if(Nota < 8){
        return 'BOM'
    }
    return 'EXCELENTE'

}


function main(){

    //entrada de dados dos alunos sexo 
    let Sexo = input('Informe o sexo do aluno[M|F]: ').toUpperCase()
    let ContadorAlunos = 0
    let SomaNotas = 0
    let ContadorHomens = 0 , ContadorMulheres = 0
    let MaiorNota = 0, MenorNota = 0
    let SomaNotasHomens = 0 , SomaNotasMulheres = 0

    while(Sexo === 'M' || Sexo === 'F'){

        ContadorAlunos++

        //entrada de dados dos alunos nota
        const Nota = GetNumberBetween('Informe a nota do aluno: ',0,10)
        SomaNotas += Nota
        
        //verificar qual é a maior e menor nota dos alunos
        if(ContadorAlunos === 1){
            MenorNota = Nota
        }

        MaiorNota = VerificarMaiorNota(MaiorNota,Nota)
        MenorNota = VerificarMenorNota(MenorNota,Nota)        
        
        //conta quantos homens e mulheres a turma tem
        if(Sexo === 'M'){
            ContadorHomens++
            SomaNotasHomens += Nota
        }else if(Sexo === 'F'){
            ContadorMulheres++
            SomaNotasMulheres +=  Nota
        }
        
        const DespenhoAluno = VerificarDesempenho(Nota)
        print(`O desempenho do aluno(a) foi [${DespenhoAluno}]`)
        
        //entrada de dados dos alunos sexo 
        Sexo = input('Informe o sexo do aluno[M|F]: ').toUpperCase()

    }

    //processamento de dados necessarios para a saída
    const MediaNotasGerais = SomaNotas / ContadorAlunos
    const MediaNotasHomens = SomaNotasHomens / ContadorHomens
    const MediaNotasMulheres = SomaNotasMulheres / ContadorMulheres
    const PerformaceHomens = ComputarPorcentagem(SomaNotasHomens,SomaNotas)
    const PerformaceMulheres = ComputarPorcentagem(SomaNotasMulheres,SomaNotas)
    const DesempenhoTurma = VerificarDesempenho(MediaNotasGerais)
    const DesempenhoHomens = VerificarDesempenho(MediaNotasHomens)
    const DesempenhoMulheres = VerificarDesempenho(MediaNotasMulheres)

    //exibição da tabela contendo todos os dados requeridos
    const TabelaFinal =`
    MAIOR NOTA GERAL                 : [${MaiorNota}]
    MENOR NOTA GERAL                 : [${MenorNota}]
    MÉDIA GERAL DA TURMA             : [${MediaNotasGerais.toFixed(1)}]
    PERFORMACE DA NOTA DOS HOMENS    : [${PerformaceHomens.toFixed(2)} %]
    PERFORMACE DA NOTA DAS MULHERES  : [${PerformaceMulheres.toFixed(2)} %]
    QUANTIDADE DE ALUNOS NA TURMA    : [${ContadorAlunos}]
    QUANTIDADE SEXO MASCULINO        : [${ContadorHomens}]
    QUANTIDADE SEXO FEMININO         : [${ContadorMulheres}]
    DESEMPENHO DA GERAL DA TURMA     : [${DesempenhoTurma}]
    DESEMPENHO HOMENS                : [${DesempenhoHomens}]
    DESEMPENHO MULHERES              : [${DesempenhoMulheres}]
    `

    print(TabelaFinal)

}

main()

File: API_SC_01_08Abril/pedro guilherme leal de araujo/q3_media.py
def questions():

    i = 0
    maior = 0
    menor = 10**200-1
    not_geral= 0
    not_homen = 0
    not_mulheres = 0
    num_mulheres = 0
    num_homens = 0

    sexo = input("Digite seu sexo (F ou M): ").upper()
    while(sexo=="F" or sexo == "M"):
        i+=1
        
        nota = int(input("Qual a nota do aluno? "))
        not_geral = not_geral+nota

        if(sexo == "F"):
            not_mulheres = not_mulheres+nota
            num_mulheres +=1
        elif(sexo == "M"):
            not_homen = not_homen+nota
            num_homens+=1

        if(nota>=maior):
            maior = nota
        if(nota<=menor):
            menor = nota
        
        sexo = input("Digite seu sexo (F ou M): ").upper()
        
    
    
        
    if(0>=not_geral/i>=2):
        desempenho_turma = "Péssimo"
    elif(2>=not_geral/i>=4):
        desempenho_turma = "Ruim"
    elif(4>=not_geral/i>=7):
        desempenho_turma = "Regular"
    elif(7>=not_geral/i>=8):
        desempenho_turma = "Bom"
    elif(8>=not_geral/i>=10):
        desempenho_turma = "Excelente"

    #print(desempenho_turma)
    
    if(num_mulheres>0 and num_mulheres>0):      
        media_geral = not_geral/i
        media_homens = not_homen/(num_homens)
        media_mulheres = not_mulheres/(num_mulheres)

    print(" ")
    print(" ")
    print(f"Há {i} alunos na turma!!")
    print(f"Há {num_mulheres-1} mulheres na turma!!")
    print(f"Há {num_homens-1} homens na turma!!")
    print(" ")
    print(f"A maior nota é {maior}")
    print(f"A menor nota é {menor}")
    print(f"A média da turma é {media_geral}")
    print(f"A performace dos homens foi de {(media_homens/media_geral)*100:.2f}% em relação a média geral")
    print(f"A performace das mulheres foi de {(media_mulheres/media_geral)*100:.2f}% em relação a média geral")
    print(f"A média dos homens é {media_homens}")
    print(f"A média dos mulhers é {media_mulheres}")
    #print(f"O desempenho da turma foi {desempenho_turma}")

questions()